tutorials/publish_subscribe/java/build.gradle

```diff
@@ -15,11 +15,35 @@ sourceSets {
 }
 
 task generateCode(type: Exec) {
+    // Idl file
+    def idlFile = 'home_automation.idl'
+    // Generated types
+    def generatedTypes = ['DeviceStatus']
+
     commandLine System.getenv('NDDSHOME') + '/bin/rtiddsgen',
         '-language', 'Java',
+        '-replace',
         'home_automation.idl'
+
     onlyIf {
-        !file('DeviceStatus.java').exists()
+        def suffixes = [
+            '.java',
+            'DataReader.java',
```

Comment:
I don't think we need to be as exhaustive. We can just compare it to one of the generated files. We can assume that the last modified time of all generated file will be the same and they won't be modified by hand.

It's unfortunate that we don't generate any file based on the idlFile name, so we have to hard-code this dependency on the name of one of the types in the file. It would be nice if we generated `home_automation-timestamp.java` (an empty file) so we can reliably associate these two files in the build system.

Let's just hardcode `DeviceStatus.java` then.